pkg/asset/machines/baremetal/hosts.go

```diff
@@ -0,0 +1,100 @@
+package baremetal
+
+import (
+	"fmt"
+
+	"github.com/sirupsen/logrus"
+
+	machineapi "github.com/openshift/cluster-api/pkg/apis/machine/v1beta1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	baremetalhost "github.com/metal3-io/baremetal-operator/pkg/apis/metal3/v1alpha1"
+
+	"github.com/openshift/installer/pkg/types"
+)
+
+// HostSettings hold the information needed to build the manifests to
+// register hosts with the cluster.
+type HostSettings struct {
+	// Hosts are the BareMetalHost objects for the hardware making up
+	// the cluster.
+	Hosts []baremetalhost.BareMetalHost
+	// Secrets holds the credential information for communicating with
+	// the management controllers on the hosts.
+	Secrets []corev1.Secret
+}
+
+// Hosts returns the HostSettings with details of the hardware being
+// used to construct the cluster.
+func Hosts(config *types.InstallConfig, machines []machineapi.Machine) (*HostSettings, error) {
+	settings := &HostSettings{}
+
+	if config.Platform.BareMetal == nil {
+		return nil, fmt.Errorf("no baremetal platform in configuration")
+	}
+
+	for i, host := range config.Platform.BareMetal.Hosts {
+		logrus.Debugf("configuring host: name = %s, details = %v\n", host.Name, host)
+
+		// Each host needs a secret to hold the credentials for
+		// communicating with the management controller that drives
+		// that host.
+		secret := corev1.Secret{
+			TypeMeta: metav1.TypeMeta{
+				APIVersion: "v1",
+				Kind:       "Secret",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      fmt.Sprintf("%s-bmc-secret", host.Name),
+				Namespace: "openshift-machine-api",
+			},
+			Data: map[string][]byte{
+				"username": []byte(host.BMC.Username),
+				"password": []byte(host.BMC.Password),
+			},
+		}
+		settings.Secrets = append(settings.Secrets, secret)
+
+		newHost := baremetalhost.BareMetalHost{
+			TypeMeta: metav1.TypeMeta{
+				APIVersion: "metal3.io/v1alpha1",
+				Kind:       "BareMetalHost",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      host.Name,
+				Namespace: "openshift-machine-api",
+			},
+			Spec: baremetalhost.BareMetalHostSpec{
+				Online: true,
+				BMC: baremetalhost.BMCDetails{
+					Address:         host.BMC.Address,
+					CredentialsName: secret.Name,
+				},
+				BootMACAddress:  host.BootMACAddress,
+				HardwareProfile: host.HardwareProfile,
+			},
+		}
+		if i < len(machines) {
+			// NOTE(dhellmann): Setting ExternallyProvisioned to
```

Comment:
who is this note for? `dhellmann`? or reader of this code?

Reply:
It's a note by me to the reader of the code. I'm used to including a name like that to aid someone trying to track down the origin of something unusual like this. Should I remove the name?

Reply:
Yeah, just drop the name, the commit has user defined.